src/pyotp/hotp.py

```diff
@@ -29,6 +29,11 @@ def __init__(
         """
         if digest is None:
             digest = hashlib.sha1
+        elif digest in [
+            hashlib.md5,
+            hashlib.shake_128
+        ]:
+            raise ValueError("selected digest function must generate digest size greater than or equals to 18 bytes")
```

Comment:
If I, for some reason (like auditing instrumentation), wrap md5 in a custom call, this won't catch it. Also, the list of overspecific; there might be more of such algorithms (in the future?), and we don't want to miss those.

It will thus be best to check the `.digest_size` attribute, as done in `otp.py:generate_otp()`.